src/main/java/io/phdata/streamliner/schemadefiner/attributeretriever/HiveAttributeRetriever.java

```diff
@@ -0,0 +1,94 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package io.phdata.streamliner.schemadefiner.attributeretriever;
+
+import io.phdata.streamliner.schemadefiner.model.FileFormat;
+import java.sql.*;
+import java.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HiveAttributeRetriever implements AttributeRetriever {
+  private static final Logger log = LoggerFactory.getLogger(HiveAttributeRetriever.class);
+  private final Connection con;
+
+  private final String HIVE_USE_DATABASE = "USE %s;";
+  private final String HIVE_DESCRIBE_TABLE = "DESCRIBE FORMATTED %s;";
+
+  private static final Map<String, String> hiveSnowflakeFileTypeMapping = new HashMap<>();
+
+  static {
+    hiveSnowflakeFileTypeMapping.put("org.apache.hadoop.mapred.TextInputFormat", "CSV");
+    hiveSnowflakeFileTypeMapping.put(
+        "org.apache.hadoop.hive.ql.io.avro.AvroContainerInputFormat", "AVRO");
+    hiveSnowflakeFileTypeMapping.put(
+        "org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat", "PARQUET");
+  }
+
+  public HiveAttributeRetriever(Connection con) {
+    this.con = con;
+  }
+
+  @Override
+  public List<Map<String, Object>> retrieve(String schema, String tableName) {
```

Comment:
'schema' should be 'database' in Hive

Reply:
If this is just to stay consistent with the rest of the project I guess it's not a big deal

Reply:
changed it to database for Hive.